source/lac/trilinos_sparse_matrix.cc

```diff
@@ -1102,7 +1099,6 @@ namespace TrilinosWrappers
   SparseMatrix::clear_rows (const std::vector<size_type> &rows,
                             const TrilinosScalar          new_diag_value)
   {
-    compress();
```

Comment:
This, too, is interesting. `clear_rows()` calls `clear_row()` for each of the rows, but `clear_row` ignores all rows not locally stored. `clear_rows()` called `compress()` at the beginning (it can't know whether any add or set operations are pending, so it doesn't know what the correct operation is) and then calls `compress(insert)` again at the end. The latter clearly makes no sense because every processor only operates on its local elements and there are no ghost rows for matrices, so there is nothing to communicate. I feel that the former is wrong because we would need to know what operation is pending but can't right here. I'll document that this is the user's responsibility.